webapp/tests/karma/ts/modules/contacts/contacts.component.spec.ts

```diff
@@ -0,0 +1,395 @@
+        sessionService = TestBed.inject(SessionService);
+        authService = TestBed.inject(AuthService);
+        contactTypesService = TestBed.inject(ContactTypesService);
+        scrollLoaderProvider = TestBed.inject(ScrollLoaderProvider);
+      });
+  }));
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('should create ContactsComponent', () => {
+    expect(component).to.exist;
+  });
+  
+  it('ngOnInit() should load and filter contacts and watch for changes', () => {
+    changesService.subscribe.reset();
+    const spySubscriptionsAdd = sinon.spy(component.subscription, 'add');
+    component.filters = {};
+
+    component.ngOnInit();
+
+    expect(searchService.search.callCount).to.equal(1);
+    expect(changesService.subscribe.callCount).to.equal(1);
+    expect(spySubscriptionsAdd.callCount).to.equal(2);
+  });
+
+  it('ngOnDestroy() should unsubscribe from observables', () => {
+    const spySubscriptionsUnsubscribe = sinon.spy(component.subscription, 'unsubscribe');
+
+    component.ngOnDestroy();
+
+    expect(spySubscriptionsUnsubscribe.callCount).to.equal(1);
+  });
+
+  describe('Search', () => {
+    it('Puts the home place at the top of the list', fakeAsync(() => {
+      searchResults = [
+        {
+          _id: 'search-result',
+        },
+      ];
+      sinon.stub(ContactsActions.prototype, 'updateContactsList');
+      searchService.search.resolves(searchResults);
+      component.contactsActions.updateContactsList = sinon.stub();
+      component.ngOnInit();
+      flush();
+      const argument = component.contactsActions.updateContactsList.args[0][0];
+
+      expect(argument.length).to.equal(2);
+      expect(argument[0]._id).to.equal('abcde');
+      expect(argument[1]._id).to.equal('search-result');
+    }));
+
+    it('Only displays the home place once', fakeAsync(() => {
+      searchResults = [
+        {
+          _id: 'search-result',
+        },
+        {
+          _id: 'abcde',
+        },
+      ];
+
+      sinon.stub(ContactsActions.prototype, 'updateContactsList');
+      searchService.search.resolves(searchResults);
+      component.contactsActions.updateContactsList = sinon.stub();
+      component.ngOnInit();
+      flush();
+      const argument = component.contactsActions.updateContactsList.args[0][0];
+
+      expect(argument.length).to.equal(2);
+      expect(argument[0]._id).to.equal('abcde');
+      expect(argument[1]._id).to.equal('search-result');
+    }));
+
+    it('Only searches for top-level places as an admin', fakeAsync(() => {
+      store.overrideSelector(Selectors.getIsAdmin, true);
+      userSettingsService.get.resolves({ facility_id: undefined });
+      getDataRecordsService.get.resolves({});
+      searchResults = [
+        {
+          _id: 'search-result',
+        },
+      ];
+      component.ngOnInit();
+      flush();
+
+      expect(contactTypesService.getChildren.args[1].length).to.equal(0);
```

Comment:
Looks like a lot of the assertions from this test were removed. 
Those that test the parameters the search service was called with or which contacts end up in the list.